include/seqan3/alignment/pairwise/policy/affine_gap_policy.hpp

```diff
@@ -58,73 +65,102 @@ class affine_gap_policy
     //!\}
 
     /*!\brief Computes the score of the current cell.
-     * \tparam  matrix_entry_type The type of the current cell.
-     * \tparam  cache_type        The type of the cache.
+     * \tparam cell_t The type of the current cell [for detailed information on the type see below].
      * \param[in,out] current_cell The current cell in the dynamic programming matrix.
      * \param[in,out] cache        The cache storing hot helper variables.
      * \param[in]     score        The score of comparing the respective letters of the first and the second sequence.
+     *
+     * \details
+     *
+     * `cell_t` is the result type of dereferencing the zipped iterator over the respective alignment score matrix and
+     * the alignment trace matrix used inside of the seqan3::detail::alignment_matrix_policy. The first parameter
+     * stored in the zipped tuple is the seqan3::detail::alignment_score_matrix_proxy and the second value is the
+     * seqan3::detail::alignment_trace_matrix_proxy.
      */
-    template <typename matrix_entry_type, typename cache_t>
-    constexpr void compute_cell(matrix_entry_type && current_cell,
-                                cache_t & cache,
+    template <typename cell_t>
+    constexpr void compute_cell(cell_t && current_cell,
+                                alignment_algorithm_cache<score_t> & cache,
                                 score_t const score) const noexcept
     {
-        using std::get;
-
-        // Unpack the cached variables.
-        auto & [score_entry, coordinate, trace_value] = current_cell;
-        auto & [main_score, hz_score, hz_trace]       = score_entry;
-        auto & [prev_cell, gap_open, gap_extend, opt] = cache;
-        auto & [prev_score, vt_score, vt_trace]       = prev_cell;
-
-        //TODO For the local alignment we might be able to use GCC overflow builtin arithmetics, which
-        // allows us to check if overflow/underflow would happen. Not sure, if this helps with the performance though.
-        // (see https://gcc.gnu.org/onlinedocs/gcc/Integer-Overflow-Builtins.html)
-
+        // score_cell = seqan3::detail::alignment_score_matrix_proxy
+        // trace_cell = seqan3::detail::alignment_trace_matrix_proxy
+        auto & [score_cell, trace_cell] = current_cell;
+        constexpr bool with_trace = !decays_to_ignore_v<std::remove_reference_t<decltype(trace_cell.current)>>;
```

Comment:
Not sure, but if it "decays to ignore" shouldn't the reference be removed by decay?